### ISSUE TITLE
Scores are -1 and no Boxes Drawn

### ISSUE THREAD

**Issue by cyrilzakka** [xuannianz/EfficientDet#222]
I think I've reached the final boss of issues: everything runs smoothly from training to inference, but resulting boxes are empty `[]` and scores are:
```
[-1. -1. -1. -1. -1. -1. -1. -1. -1. -1. -1. -1. -1. -1. -1. -1. -1. -1.
 -1. -1. -1. -1. -1. -1. -1. -1. -1. -1. -1. -1. -1. -1. -1. -1. -1. -1.
 -1. -1. -1. -1. -1. -1. -1. -1. -1. -1. -1. -1. -1. -1. -1. -1. -1. -1.
 -1. -1. -1. -1. -1. -1. -1. -1. -1. -1. -1. -1. -1. -1. -1. -1. -1. -1.
 -1. -1. -1. -1. -1. -1. -1. -1. -1. -1. -1. -1. -1. -1. -1. -1. -1. -1.
 -1. -1. -1. -1. -1. -1. -1. -1. -1. -1.]
```

I've seen some issues stating that using TF 1.15 solves this issue, but I've had no such luck

**Comment by QuietWoods** [comment]
When you finish training, convert the multi-GPU training model to a single-GPU model and save it (for inference):
` 
# build model
    single_model, prediction_model = efficientdet(args.phi,
                                           num_classes=num_classes,
                                           num_anchors=num_anchors,
                                           weighted_bifpn=args.weighted_bifpn,
                                           freeze_bn=args.freeze_bn,
                                           detect_quadrangle=args.detect_quadrangle
                                           )
    # freeze backbone layers
    if args.freeze_backbone:
        # 227, 329, 329, 374, 464, 566, 656
        for i in range(1, [227, 329, 329, 374, 464, 566, 656][args.phi]):
            single_model.layers[i].trainable = False

    if args.gpu and len(args.gpu.split(',')) > 1:
        model = keras.utils.multi_gpu_model(single_model, gpus=list(range(len(args.gpu.split(',')))))

    # compile model
    model.compile(optimizer=Adam(lr=1e-3), loss={
        'regression': smooth_l1_quad() if args.detect_quadrangle else smooth_l1(),
        'classification': focal()
    }, )

    # load pretrained weights
    if args.snapshot:
        if args.snapshot == 'imagenet':
            raise ValueError("When you converter model, dont use imagenet")
        else:
            print('Loading model, this may take a second...')
            model.load_weights(args.snapshot, by_name=True, skip_mismatch=True)

    # create the callbacks
    callbacks = create_callbacks(
        model,
        prediction_model,
        validation_generator,
        args,
    )
    
    print("Saving device model, this may take a second...")
    model_name = "device_" + os.path.basename(args.snapshot)
    out_model_path = os.path.join(args.snapshot_path, model_name)
    single_model.save_weights(out_model_path)
    print("Save success")`

**Comment by eypros** [comment]
The results are being tested on an unknown image for example. I would recommend if so to check it against a training set image sample which increase the probability to actually get some results. -1 generally means no bboxes were found. This could be an error in code but also it just could mean the provided image does not contain any bbox (especially if the threshold has been set to a high value, e.g. 0.5).
